kong/plugins/rate-limiting/handler.lua

```diff
@@ -105,25 +111,61 @@ function RateLimitingHandler:access(conf)
     end
   end
 
+  local timestamps
+  local reset
   if usage then
     -- Adding headers
     if not conf.hide_client_headers then
       local headers = {}
+      -- Find the closer limit.
+      local lowest = nil
       for k, v in pairs(usage) do
         if stop == nil or stop == k then
           v.remaining = v.remaining - 1
         end
 
+        local remaining = max(0, v.remaining)
+
         headers[RATELIMIT_LIMIT .. "-" .. k] = v.limit
-        headers[RATELIMIT_REMAINING .. "-" .. k] = max(0, v.remaining)
+        headers[RATELIMIT_REMAINING .. "-" .. k] = remaining
+
+        if stop and remaining == 0 then
+          if not timestamps then
+            timestamps = timestamp.get_timestamps(current_timestamp)
+          end
+
+          reset = max(reset or 0, EXPIRATIONS[k] - floor((current_timestamp -
+                                                          timestamps[k]) / 1000))
+        end
+
+        -- find the lowest limit in terms of remaining calls or time-window
+        if (lowest == nil
+            or usage[lowest].remaining > remaining
+            or (usage[lowest].remaining == remaining and EXPIRATIONS[k] > EXPIRATIONS[lowest])
+           ) then
+          lowest = k
+        end
       end
 
+      -- Set the normalized ratelimit values.
+      headers[RATELIMIT_LIMIT] = headers[RATELIMIT_LIMIT .. "-" .. lowest]
+      headers[RATELIMIT_REMAINING] = headers[RATELIMIT_REMAINING .. "-" .. lowest]
+      -- Set the expected reset time.
+      headers[RATELIMIT_RESET] = EXPIRATIONS[lowest] - (current_timestamp/1000 % EXPIRATIONS[lowest])
```

Comment:
According to:
https://tools.ietf.org/html/draft-polli-ratelimit-headers-01

> If a response contains both the "Retry-After" and the "RateLimit- Reset" header fields, the value of "RateLimit-Reset" MUST be  consistent with the one of "Retry-After".

So I don't think it is a good idea to have `RateLimit-Reset` or `X-RateLimit-Reset` being something else than `Retry-After` or inconsistent with it.

But having specific:
`X-RateLimit-Reset-Minute` (etc.) could be an option. What do you think?

I know that the `polli` rfc is just a draft but that is the best we have. Everything else is proprietary, except perhaps `Retry-After` on `429`. And yes, we should drop `X-` prefix at some point (not now).

Reply:
IIUC in this implementation `RateLimit-Reset` and `Retry-After` should match,
but let me know if I'm wrong.

> having specific `x-ratelimit-reset-minute` ...
Header suffixes (eg `-minute`) require maps/code to interpret suffixes (Eg.  `policies.EXPIRATIONS`),
so clients  a great fan.

Can you make an example of how you would use `X-RateLimit-Reset-Minute` & co?

Thanks for your support! :D